src/Grpc.Net.Client/Balancer/BalancerAddress.cs

```diff
@@ -69,5 +71,21 @@ public override string ToString()
     {
         return $"{EndPoint.Host}:{EndPoint.Port}";
     }
+
+    private sealed class BalancerEndPoint : DnsEndPoint
+    {
+        private string? _cachedToString;
+
+        public BalancerEndPoint(string host, int port) : base(host, port)
+        {
+        }
+
+        public override string ToString()
+        {
+            // Improve ToString performance when logging by caching ToString.
+            // Don't include DnsEndPoint address family.
+            return _cachedToString ??= $"{Host}:{Port}";
```

Comment:
I assume, from context, that this is logged _a lot_?

Reply:
Not that much. But it's simple enough to cache.